### ISSUE TITLE
[feat] recursive option for fileDropEnabled

### ISSUE THREAD

**Issue by mauritzn** [tauri-apps/tauri#6376]
### Describe the problem

It would be nice to have a `recursive` option for `fileDropEnabled`, so that it would work the same that `dialog.open` works with the `recursive` option. My application needs to access files inside sub directories in the provided folder, but dropped items only seem to add the main folder and first level of files to the fs scope.

### Describe the solution you'd like

I'm not quite sure how this option would be implemented, an additional configuration setting of `fileDropRecursive` in `tauri.conf.json` could be added.

**e.g.**
```json5
{
  "tauri": {
    // ...
    "windows": [
      {
        "title": "Tauri App",
        "width": 1600,
        "height": 900,
        "resizable": true,
        "fullscreen": false,
        "fileDropRecursive": true
      }
    ]
  }
}
```

### Alternatives considered

I'm currently forced to have a global fs scope _(e.g. `["**"]`)_, which is something I'd very much like to avoid. I tried to figure out how to extend the fs scope through Rust, but even after hours of scouring through GitHub issues/discussions, the Tauri website and the Rust docs for Tauri _(even tried ChatGPT and Bing Chat... I got desperate...)_, I wasn't able to figure out how to do it...

Tauri is my first interaction with Rust and it's quite hard figuring out the Rust docs when no examples are provided for how to use the methods listed in the documentation, on the other hand the provided JavaScript/TypeScript documentation shows nice minimal examples on basic usage _(would be nice if the Rust docs had similar examples for Rust beginners like me)_.

Even though I'm able to find the functions I'm suppose to use in Rust using the docs, I have no idea how to use them, everything I've tried just gives me errors _(which is why some minimal examples would be nice)_...

**Example from Rust docs:**
![Screenshot 2023-03-01 033043](https://user-images.githubusercontent.com/11979966/222021754-0b74e028-d994-4d55-8320-d5770caa02dc.png)

Sorry I got a bit of topic here... Simply put, implementing a `fileDropRecursive` configuration setting would be quite helpful and a better solution than having to manually do recursive fs scope addition from the frontend using as Rust command.

**Comment by FabianLars** [comment]
The rust solution would look something like this
```rs
#[tauri::command]
fn extend_scope(handle: tauri::AppHandle, path: std::path::PathBuf) {
  let _ = handle.fs_scope().allow_directory(path, true); //just ignoring the error here because i'm lazy
}
```

And then it would be invoked in js as described here https://tauri.app/v1/guides/features/command

Of course if you can handle the whole file drop event in rust in the first place this would be a bit more secure (there can't be random paths injected then)

**Comment by mauritzn** [comment]
Thank you @FabianLars very much for giving me a clear example. It still gave me an error since `use tauri::Manager;` wasn't included, but that was an easy fix, I cannot thank you enough, since this finally gives me feature parity with `dialog.drop`.

Sadly I don't think I'll ever manage to do the whole thing in Rust, since I need to do a fair amount of parsing of the dropped folder. The best I could implement was to handle the file drop detection in Rust, extend the scope and then emit an event to the frontend with the dropped path.

While I do now have it working, it would still be really nice to have a configuration option for this, since it's quite a bit of added code. `dialog.drop` supports it so feature parity with it would be appreciated.

Here is the end result if anyone else wants to implement something similar _(it's probably not optimal)_:
```rust
use tauri::Manager;

// the payload type must implement `Serialize` and `Clone`.
#[derive(Clone, serde::Serialize)]
struct Payload {
  path: String,
}

fn main() {
  tauri::Builder::default()
    .on_window_event(|event| match event.event() {
      tauri::WindowEvent::FileDrop(drop_event) => {
        match drop_event {
          tauri::FileDropEvent::Dropped(dropped) => {
            // only the first path is fetched
            let dropped_path = dropped.first();
            if dropped_path.is_none() == false {
              println!(
                "Scope extended with path (from file drop): {:?}",
                dropped_path.unwrap()
              );

              let window = event.window();
              let app_handle = window.app_handle();
              let fs_scope = app_handle.fs_scope();
              let asset_scope = app_handle.asset_protocol_scope();

              // INFO: error handling is not implemented here yet

              // extend fs scope
              let _fs_result = fs_scope.allow_directory(dropped_path.unwrap(), true);

              // extend asset scope
              let _asset_result = asset_scope.allow_directory(dropped_path.unwrap(), true);

              let _event_result = window.emit(
                "file-dropped-scoped",
                Payload {
                  path: dropped_path.unwrap().to_string_lossy().into(),
                },
              );
            }
          }
          _ => {}
        }
      }
      _ => {}
    })
    .invoke_handler(tauri::generate_handler![])
    .run(tauri::generate_context!())
    .expect("error while running tauri application");
}
```

**Comment by FabianLars** [comment]
> While I do now have it working, it would still be really nice to have a configuration option for this

I agree. That's why i didn't close this issue with my previous comment😅
